### PR TITLE
fix: add missing type definitions

### PR DESCRIPTION
**Pull request by andrewszucs**
**Summary**

I've run into some Typescript errors while using the Manual Initialization in a separate `.js` files, for example missing property definitions on the `CmsCollection` type. I've added the ones I used and the ones I have found described in the docs.

**Test plan**

Ran `yarn type-check` and `yarn test` without any errors.

### CODE REVIEW

File: packages/netlify-cms-core/index.d.ts [hunk 1]
```diff
@@ -23,7 +23,20 @@ declare module 'netlify-cms-core' {
...
+
+  export type CmsFilesExtension =
+    | 'yml'
+    | 'yaml'
+    | 'toml'
+    | 'json'
+    | 'md'
+    | 'markdown'
+    | 'html'
+    | string;
```
Comment (erezrokah) [inline packages/netlify-cms-core/index.d.ts:39]:
```suggestion
  export type CmsFilesExtension = string;
```
I think we can safely change this to a `string`

File: packages/netlify-cms-core/index.d.ts [hunk 2]
```diff
@@ -46,6 +59,69 @@ declare module 'netlify-cms-core' {
     label?: string;
     widget?: string;
     required?: boolean;
+    hint?: string;
```
Comment (erezrokah) [inline packages/netlify-cms-core/index.d.ts:62]: This is 🔥

File: packages/netlify-cms-core/index.d.ts (hunk 1 above)
Comment (andrewszucs) [reply packages/netlify-cms-core/index.d.ts]: Thanks, @erezrokah! I've updated it.

**Comment by erezrokah** [comment]
Thanks again @andrewszucs!

**Pull request merged**